core/base/document.php

```diff
@@ -1783,6 +1788,59 @@ protected function save_settings( $settings ) {
 	 * @access protected
 	 */
 	protected function print_elements( $elements_data ) {
+		if ( ! Plugin::$instance->experiments->is_feature_active( 'e_element_cache' ) ) {
+			$this->do_print_elements( $elements_data );
+
+			return;
+		}
+
+		$cached_data = $this->get_document_cache();
+
+		if ( false === $cached_data ) {
+			add_filter( 'elementor/element/should_render_shortcode', '__return_true' );
+
+			global $wp_scripts, $wp_styles;
+			$scripts_ignored = $wp_scripts->queue;
+			$styles_ignored = $wp_styles->queue;
+
+			ob_start();
+
+			$this->do_print_elements( $elements_data );
+
+			$scripts_to_queue = array_values( array_diff( $wp_scripts->queue, $scripts_ignored ) );
+			$styles_to_queue = array_values( array_diff( $wp_styles->queue, $styles_ignored ) );
```

Comment:
@KingYes I might be reading this wrong, but it looks like these arrays will always be empty? 🤔

Reply:
@KingYes As discussed on Slack, this seems to be causing the PlayWright failure:

This is causing the PlayWright failure:
```
Fatal error: Uncaught TypeError: array_diff(): Argument #1 ($array) must be of type array, null given in /var/www/html/wp-content/plugins/build/core/base/document.php:1812
230
Stack trace:
231
#0 /var/www/html/wp-content/plugins/build/core/base/document.php(1812): array_diff(NULL, NULL)
232
#1 /var/www/html/wp-content/plugins/build/core/base/document.php(1195): Elementor\Core\Base\Document->print_elements(Array)
233
#2 /var/www/html/wp-content/plugins/build/includes/frontend.php(1157): Elementor\Core\Base\Document->print_elements_with_wrapper(Array)
```